src/test/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseCreationReceiverImplIT.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.ons.ctp.response.casesvc.endpoint;
+
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.github.tomakehurst.wiremock.extension.responsetemplating.ResponseTemplateTransformer;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import java.util.UUID;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.embedded.LocalServerPort;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandlingException;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.response.casesvc.CaseCreator;
+import uk.gov.ons.ctp.response.casesvc.message.sampleunitnotification.SampleUnitParent;
+import uk.gov.ons.ctp.response.casesvc.service.CaseService;
+
+@ContextConfiguration
+@ActiveProfiles("test")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@RunWith(SpringJUnit4ClassRunner.class)
+public class CaseCreationReceiverImplIT {
+
+  @LocalServerPort private int port;
+
+  @Autowired private CaseCreator caseCreator;
+
+  @Autowired private MessageChannel caseTransformed;
+
+  @Rule
+  public WireMockRule wireMockRule =
+      new WireMockRule(options().extensions(new ResponseTemplateTransformer(false)).port(18002));
+
+  @MockBean private CaseService caseService;
+
+  @Test
+  public void ensureFiniteRetriesOnFailedCaseNotification()  {
```

Comment:
This test doesn't have a "then" part in the given-when-then expected structure

Reply:
We can't add one since there's no concrete way of testing for retries. You just have to observe the console output and ensure the same error appears more than once.